ext/storage_sources/s3_store/test/test_list_buckets.cpp

```diff
@@ -0,0 +1,63 @@
+#include <aws_bucket_conn.h>
+#include <aws/core/Aws.h>
+
+#include <iostream>
+
+/* Default config settings for the S3CrtClient. */
+namespace test_defaults 
+{
+    const Aws::String region = Aws::Region::AP_SOUTHEAST_2;
+    const double throughput_target_gbps = 5;
+    const uint64_t part_size = 8 * 1024 * 1024; /* 8 MB. */
+}
+
+int test_list_buckets(const Aws::S3Crt::ClientConfiguration &config);
+
+/* Wrapper for unit test functions. */
+#define TEST(func, config, expected)                   \
+    do {                                               \
+        int __ret;                                     \
+        if ((__ret = (func(config))) != expected)      \
+            return (__ret);                            \
+    } while (0)
+
+/*
+ * test_list_buckets --
+ *     Unit test for listing S3 buckets under the associated AWS account.
+ */
+int 
+test_list_buckets(const Aws::S3Crt::ClientConfiguration &config) {
+    aws_bucket_conn conn(config);
+    std::vector<std::string> buckets;
+    if (!conn.list_buckets(buckets))
+        return 1;
+             
+    std::cout << "All buckets under my account:" << std::endl;
```

Comment:
We may want to clear all the `std::cout` calls that were used for debugging.

Reply:
I think we will like to keep the std::cout calls for now to not only check if the function is working, but also to verify the connection to the AWS Api and see the changes without having to open the console.

Reply:
Let's keep them for now, we will probably remove them later on. It would be great to have a verbosity functionality where we can run the tests with and without debugging messages. Can we create a ticket for that purpose? Something along the lines of "Investigate how to improve the debugging messages". This ticket will remind us of checking this at a later stage.

Reply:
The messaging/debugging will integrate into WiredTiger. We have a ticket for that - WT-8662